include/model.php

```diff
@@ -154,7 +161,11 @@ public function get_languages_list( $args = array() ) {
 			 * @param PLL_Language[] $languages The list of language objects.
 			 */
 			$languages = apply_filters( 'pll_after_languages_cache', $languages );
-			$this->cache->set( 'languages', $languages );
+
+			if ( $this->are_languages_ready() ) {
+				$this->cache->set( 'languages', $languages );
+			}
```

Comment:
Wouldn't it be appropriate to fire a `_doing_it_wrong()` in the other case?

Reply:
Done in [cc9572f](https://github.com/polylang/polylang/pull/1211/commits/cc9572f3ce0b03a544da6b41721a622a0edbfebb).